### PR TITLE
fix: connectivity banner height

### LINKED ISSUE #9803: Connectivity banner cuts off text when message is too long
# Bug Report

## Problem

When the connectivity banner contains more than one line of copy, the text overflows. 

image

#### Expected behavior

Room for two lines of copy on the top banner.

#### Actual behavior

Bottom line of text is cut off.

### PR DESCRIPTION
**Pull request by emizzle**
Fixes #9803 

### Summary
The height of the banner was cutting off banner messages spanning multiple lines

### Fix
Due to the animation of the view surrounding the banner text, calculate the banner height, then re-render the banner with the updated height.

### Details
Render a view that wraps not only the message banner itself, but also a hidden (0-opacity and absolutely positioned) view with the banner text that calculates the height of the banner during the `on-layout` event. The banner height calculation causes the reagent atom value change to re-render the banner view with the udpated height.

### Review notes
1. There is an animation on the message banner. The "out" animation (ie when the banner is in the process of disappearing) seems to work correctly. However, there did not seem to be an "in" animation. I'm fairly confident these changes did not break any existing animations, however it's worth noting this so that others who are more familiar with the existing codebase may be in a better place to say otherwise.

#### Areas that maybe impacted
Any chat screen (chat list, 1-on-1, chat group, etc).

### Steps to test
1. Open Status in the emulator/simulator.
2. In the REPL, force the banner to appear by dispatching the `network-info-changed` event like so:
```
(re-frame/dispatch [::network-info-changed {:isConnected false}])
```
3. If you'd like to see a multi-line message, edit line `1586` in `src/status_im/subs.cljs` to add a long message, ie `:t/mailserver-request-error-status` can be used.

status: ready

Co-authored-by: Pascal Precht <pascal@status.im>
Co-authored-by: Eric Dvorsak <eric@status.im>

### CODE REVIEW

**Comment by flexsurfer** [comment]
hey @emizzle great job! could you please remove all commented lines, and revert back all formatting, thanks

**Comment by emizzle** [comment]
@flexsurfer - comments removed. I've asked in #desktop discord if we have a standardised `:cljfmt` config definition as the config in `/project.cli` does not seem to be formatting correctly. Do you know if there is something like this available? If not, what are you using to format?

File: src/status_im/ui/components/connectivity/view.cljs [hunk 1]
```diff
@@ -78,100 +73,117 @@
...
+          (reset! to-hide? true)
+          (reset! status-hidden false))))))
 
 (defn connectivity-status
-  [{:keys [connected?]} status-hidden]
-  (let [anim-translate-y (animation/create-value neg-connectivity-bar-height)
-        anim-opacity     (animation/create-value 0)]
+  [{:keys [connected?]} status-hidden height]
+  (let [neg-connectivity-bar-height (- @height)
+        anim-translate-y            (animation/create-value neg-connectivity-bar-height);(- (or connectivity-bar-height-message connectivity-bar-height)));neg-connectivity-bar-height)
```
Comment (flexsurfer) [inline src/status_im/ui/components/connectivity/view.cljs]: also here

File: src/status_im/ui/components/connectivity/view.cljs [hunk 2]
```diff
@@ -78,100 +73,117 @@
...
+                                                     :opacity  0})
+                      :on-layout (fn [e]
+                                   (let [h (-> e .-nativeEvent .-layout .-height)]
+                                     (reset! height (+ h (* (:top styles/text) 2)))))}
+          (i18n/label (:message props))])
+       [connectivity-status props status-hidden height]])))
 
 ;; timer updating the enqueued status
+
+
```
Comment (flexsurfer) [inline src/status_im/ui/components/connectivity/view.cljs]: extra lines

File: src/status_im/ui/components/connectivity/view.cljs [hunk 3]
```diff
@@ -34,34 +31,32 @@
...
-           (animation/parallel
-            [(animation/timing blue-bar-left-margin (easing :out 0.85))
-             (animation/timing white-bar-left-margin (easing :out 0.85))])
-           (animation/parallel
-            [(animation/timing blue-bar-left-margin (easing :in 0.19))
-             (animation/timing white-bar-left-margin (easing :in 0.65))])
-           (animation/parallel
-            [(animation/timing blue-bar-left-margin (easing :out 0))
-             (animation/timing white-bar-left-margin (easing :out 0))])]))))}
+    {:component-did-mount (fn [_]
```
Comment (flexsurfer) [inline src/status_im/ui/components/connectivity/view.cljs]: is this formatting also made by a tool?

Comment (emizzle) [reply src/status_im/ui/components/connectivity/view.cljs]: Yes, it is formatted by `cljfmt` using the config found here: https://github.com/status-im/status-react/blob/de4e73a2566abd647f97d27dd195404be379b9bf/project.clj#L32

File: src/status_im/ui/components/connectivity/view.cljs (hunk 2 above)
Comment (emizzle) [reply src/status_im/ui/components/connectivity/view.cljs]: Removed.

File: src/status_im/ui/components/connectivity/view.cljs (hunk 1 above)
Comment (emizzle) [reply src/status_im/ui/components/connectivity/view.cljs]: Removed.

File: src/status_im/ui/components/connectivity/view.cljs [hunk 4]
```diff
@@ -78,98 +73,113 @@
...
-             [react/nested-text {:style    styles/text
-                                 :on-press (when on-press-event #(re-frame/dispatch [on-press-event]))}
-              (i18n/label :t/waiting-for-wifi) " "
-              [{:style {:text-decoration-line :underline}}
-               (i18n/label :t/waiting-for-wifi-change)]]
-             (when message
-               [react/text {:style    styles/text
-                            :on-press (when on-press-event #(re-frame/dispatch [on-press-event]))}
-                (i18n/label message)]))]))})))
+      :component-did-update    (fn [comp]
```
Comment (yenda) [inline src/status_im/ui/components/connectivity/view.cljs]: but cljfmt doesn't do that, we try to keep the length of the lines under 80 chars (soft limit), for this reason in cases like this we put the key and value on separate lines

File: src/status_im/ui/components/connectivity/view.cljs (hunk 3 above)
Comment (emizzle) [reply src/status_im/ui/components/connectivity/view.cljs]: Fixed.

**Review by yenda — APPROVED** [review]

**Comment by churik** [comment]
Failed e2e tests are not related, they are already fixed in develop

**Comment by churik** [comment]
Well, I couldn't trigger 2-lines connectivity banner, but 1-lines looks OK to me on Android and IOS.
@flexsurfer WDYT?

**Comment by flexsurfer** [comment]
merged

**Pull request closed without merging**